libraries/ESP8266WebServer/examples/FSBrowser/readme.md

```diff
@@ -0,0 +1,121 @@
+# FSBrowser readme
+
+## What is this sketch about ?
+
+This example is a FileSystem Browser for the ESP8266 using http requests and a html/javascript frontend, 
+working for both SPIFFS, LittleFS and SDFS.
+This unified version is based on the previous examples named FSWebServer, FSBrowser and SDWebServer, Copyright (c) 2015 Hristo Gochkov. All rights reserved.
+
+## How to use ?
+1. Uncomment one of the `#define USE_xxx` directives in the sketch
+2. Add the credentials of your WiFi network (search for `STASSID`)
+3. Compile and upload the sketch to your ESP8266 device
+4. For normal use, copy the contents of the `data` folder to the filesystem. To do so:
+- for SDFS, copy that contents (not the data folder itself, just its contents) to the root of a FAT/FAT32-formated SD card connected to the SPI port of the ESP8266
+- for SPIFFS or LittleFS, please follow the instructions at https://arduino-esp8266.readthedocs.io/en/latest/filesystem.html#uploading-files-to-file-system
+5. Once the data and sketch have been uploaded, access the editor by pointing your browser to http://fsbrowser.local/edit
```

Comment:
/edit or /edit.htm?

Reply:
It's `edit`.
`edit` is the folder, and the default file is `index.htm`.

Reply:
Is /edit then both a service endpoint and a folder? How does that work?

Reply:
Well, from the user's point of view, you call `/edit` URL from your browser, and you get the `index.htm` file from the `/edit` folder.
Now admittedly `/edit` has specific handlers for the different HTTP methods, and when called in GET, the dedicated handler optionally returns the embedded gzipped version if the `index.htm` file is not found, but functionally, the user must still call `http://fsbrowser.local/edit` to open the FS Browser page.